back/engines/commercial/insights/app/finders/insights/inputs_finder.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Insights
+  class InputsFinder
+
+    MAX_PER_PAGE = 100
+
+    attr_reader :view, :params
+
+    def initialize(view, params = {})
+      @view = view
+      @params = params
+    end
+
+    def execute
+      inputs = view.scope.ideas
+      inputs = filter_category(inputs)
+      inputs = search(inputs)
+      paginate(inputs)
+    end
+
+    # Takes into account, both, actual and suggested categories.
+    # Keep only inputs without categories if +params[:category]+ is +nil+ 
+    # or +''+.
+    # @raise [ActiveRecord::RecordNotFound]
+    def filter_category(inputs)
+      return inputs unless (category_id = params[:category])
```

Comment:
Doesn't this return all inputs when `params[:category]` is nil ?

Reply:
But then isn't the comment misleading ?

Reply:
Ooooh good catch! 👀 I tried a few variants and I ended up mixing two of them here. Fixed.